app/models/journal/aggregated_journal.rb

```diff
@@ -0,0 +1,168 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+# Similar to regular Journals, but under the following circumstances journals are aggregated:
+#  * they are in temporal proximity
+#  * they belong to the same resource
+#  * they were performed by the same user
+#  * no other user has an own journal on the same object between the aggregated ones
+# When a user commented (added a note) twice within a short time, the second comment will
+# "open" a new aggregation, since we do not want to merge comments in any way.
+# The term "aggregation" means the following when applied to our journaling:
+#  * ignore/hide old journal rows (since every journal row contains a full copy of the journaled)
+#    object, dropping intermediate rows will just increase the diff of the following journal
+#  * in case an older row had notes, take the notes from the older row, since they shall not
+#    be dropped
+class Journal::AggregatedJournal < Journal
+  self.table_name = 'journals'
+
+  class << self
+    def default_scope
+      # Using the roughly aggregated groups from :sql_rough_group we need to merge journals
+      # where an entry with empty notes follows an entry containing notes, so that the notes
+      # from the main entry are taken, while the remaining information from the more recent entry
```

Comment:
Suggestion: "..., while the remaining information is taken from the more recent entry."